### DISCUSSION TITLE
[question] How to use create-exe for my game?

### DISCUSSION THREAD

**Discussion by paulocoutinhox** [wasmerio/wasmer#3112] [Q&A]
### Summary

Hi,

Im reading your article about universal binary format. How can i convert my open source game example to this executable?

How can i use create-exe for my game?

The .wasm file if here:
https://github.com/nativium/nativium-game/actions/runs/2847623683

Or attached:
[nativium.wasm.zip](https://github.com/wasmerio/wasmer/files/9330921/nativium.wasm.zip)

Today im getting error:

```
paulo ~/Downloads/wasmer  $ wasmer create-exe -o my-exe nativium.wasm
Compiler: cranelift
Target: aarch64-apple-darwin
Format: Symbols
Using libwasmer file: /Users/paulo/.wasmer/lib/libwasmer.a
Using path `/Users/paulo/.wasmer/lib/libwasmer.a` as libwasmer path.
✔ Native executable compiled successfully to `my-exe`.
paulo ~/Downloads/wasmer  $ ./my-exe                                 
Error getting WASI imports!
Error len: `41`
The WASI version could not be determined
```

**Comment by jcaesar** [comment]
The binary may be architecture-independent, but it still depends on the environment it was compiled for (which is the specific environment provided by `nativium.js` from the same tar). You can have a look at this with
```plain
$ wasmer inspect release/wasm/nativium.wasm
Type: wasm
Size: 978.7 KB
Imports:
  Functions:
    "a"."a": [] -> [I32]
    "a"."b": [I32] -> []
    "a"."c": [I32] -> []
    "a"."d": [I32] -> []
    "a"."e": [I32, I32, I32] -> []
    "a"."f": [I32, I32, I32] -> [I32]
…
```
This looks to be a minified and recent version of the emscripten environment, which isn't supported by wasmer.  (As indicated by the fact that it's looking for WASI.)

In general, I wouldn't expect to be able to compile and run arbitrary things with WASM. Games are on the more difficult side. [Experimental support for GUI](https://github.com/wasmerio/wasmer/tree/master/lib/wasi-experimental-io-devices)s exists, but iirc no sound. If you want to make a standalone version of a game that originally targets the web, using electron or tauri instead of wasmer might be a more walkable path.
